lib/Pipeline.js

```diff
@@ -139,6 +139,24 @@ class Pipeline {
     })
   }
 
+  async parseArguments (args) {
+    // is it a list?
+    if (args.out(ns.rdf.first).terms.length === 1) {
+      return Promise.all([...args.list()].map(arg => this.parseArgument(arg)))
+    }
+
+    // merge all key value pairs into an object
+    const argObject = await args.toArray().reduce(async (p, argNode) => {
+      const argsResolved = await p
+
+      argsResolved[argNode.out(ns.p.name).value] = await this.parseArgument(argNode.out(ns.p.value))
```

Comment:
I guess these don't need to be awaited sequentially, not sure awaiting them concurrently is worth the effort though.

Reply:
Yes, I do think that order is irrelevant. Do you have an improvement in mind?

Reply:
Yep, here is an example where `sequential.js` is what `argObject` does here and `concurrent.js` is the same but resolving concurrently: https://gist.github.com/vhf/26785cbca3b3cd56b42e1b3ae9c302cf

Reply:
before
```js
// merge all key value pairs into an object
const argObject = await args.toArray().reduce(async (p, argNode) => {
  const argsResolved = await p

  argsResolved[argNode.out(ns.p.name).value] = await this.parseArgument(argNode.out(ns.p.value))

  return argsResolved
}, Promise.resolve({}))
```
after
```js
const argNodes = args.toArray()
const promises = argNodes.map((argNode) => this.parseArgument(argNode.out(ns.p.value)))
const values = await Promise.all(promises)
// merge all key value pairs into an object
const argObject = argNodes.reduce((acc, argNode, idx) => {
  acc[argNode.out(ns.p.name).value] = values[idx]
  return acc
}, {})
```

Reply:
I changed it to @vhf proposal.